mains/rdas_fv3jedi.cc

```diff
@@ -0,0 +1,108 @@
+// -------------------------------------------------------------------------------------------------
+
+#include <functional>
+#include <map>
+
+#include "fv3jedi/ObsLocalization/instantiateObsLocFactory.h"
+#include "fv3jedi/Utilities/Traits.h"
+
+#include "oops/generic/instantiateModelFactory.h"
+#include "saber/oops/instantiateCovarFactory.h"
+#include "saber/oops/ErrorCovarianceToolbox.h"
+#include "ufo/instantiateObsErrorFactory.h"
+#include "ufo/instantiateObsFilterFactory.h"
+#include "ufo/ObsTraits.h"
+
+#include "oops/runs/ConvertState.h"
+#include "oops/runs/HofX4D.h"
+#include "oops/runs/LocalEnsembleDA.h"
+#include "oops/runs/Run.h"
+#include "oops/runs/Variational.h"
+
+// -------------------------------------------------------------------------------------------------
+
+template<typename Traits>
+int runApp(int argc, char** argv, const std::string appName) {
+
+  // Create the Run object
+  oops::Run run(argc, argv);
+
+  // Instantiate oops factories
+  oops::instantiateModelFactory<Traits>();
+
+  // Instantiate saber factories
+  saber::instantiateCovarFactory<Traits>();
+
+  // Intantiate ufo factories
+  ufo::instantiateObsErrorFactory();
+  ufo::instantiateObsFilterFactory();
+
+  // Localization for ensemble DA
+  if (appName == "localensembleda") {
+    fv3jedi::instantiateObsLocFactory();
+  }
+
+  // Application pointer
+  std::unique_ptr<oops::Application> app;
+
+  // Define a map from app names to lambda functions that create unique_ptr to Applications
+  std::map<std::string, std::function<std::unique_ptr<oops::Application>()>> apps;
+
+  apps["convertstate"] = []() {
```

Comment:
you may want the new "converttostructuredgrid" to make lat/lon plotting easier

Reply:
Thank you Cory for the quick review! Unfortunately the oops hash linked in RDASApp is from [four months ago](https://github.com/jcsda/oops/tree/f02f283b7c4ea1edae349413964f3edf0f7fab33), before the ConvertToStructuredGrid app was added. I don't think we have a made a decisions about when or how often to update these. Is it an easy change to update?

Reply:
ah good point, it may be non-trivial to update everything. I would suggest doing that in a future PR.